benchadapt/python/benchadapt/machine_info.py

```diff
@@ -77,6 +77,33 @@ class GitParseWarning(RuntimeWarning):
 
 
 def github_info():
+    """Get github metadata from environment variables"""
+    repo = os.environ.get("CONBENCH_REPOSITORY")
+    pr_number = os.environ.get("CONBENCH_PR_NUMBER") or os.environ.get(
+        "BENCHMARKABLE_PR_NUMBER"
+    )
+    commit = os.environ.get("CONBENCH_COMMIT")
+
+    gh_info = {
+        "commit": commit,
+        "repository": repo,
+        # "branch": None,
+        "pr_number": pr_number,
+    }
+
+    if not (repo and pr_number and commit):
+        warnings.warn(
+            (
+                "All of CONBENCH_REPOSITORY, CONBENCH_PR_NUMBER, and "
+                "CONBENCH_COMMIT must be set if `github` is not specified. "
+                f"\nValues: `{gh_info}`"
+            ),
+        )
```

Comment:
> I also sorta prefer an error here before we post a messy `{"commit": None, "repository": None, "pr_number": None}` to Conbench. But I understand the "adapter override" behavior makes this hard. Can we error out somewhere else?

Because of that we really can't on object creation. We do have [some similar warnings on the `.to_publishable_dict()` method](https://github.com/conbench/conbench/blob/main/benchadapt/python/benchadapt/result.py#L160-L168). We could move this warning there, though here is better because it will also get raised in other places `machine_info()` gets called. We _might_ be able to escalate the warning to an error there? That would cause augmentation in benchconnect to error out if they're not set, which is probably desirable. I'll try it and see what I break.

Reply:
Ok, made `.to_publishable_dict()` error on run and result if commit and repo aren't specified and fixed what broke. I'm now wondering if the other warnings there could be errors too; I think I tried and then relaxed at some point, but I forget why